lib/robot-controller/robots.rb

```diff
@@ -1,87 +1,93 @@
 require 'yaml'
 
 #
-class RobotConfigParser
-  ROBOT_INSTANCE_MAX = 16
+module RobotController
+  #
+  class Parser
+    ROBOT_INSTANCE_MAX = 16
 
-  # parse_instances(1) == 1
-  # parse_instances(16) == 16
-  # parse_instances(0) == 1
-  # parse_instances(99) => RuntimeError
-  def parse_instances(n)
-    if n > ROBOT_INSTANCE_MAX
-      fail "TooManyInstances: #{n} > #{ROBOT_INSTANCE_MAX}"
-    end
-    n = 1 if n < 1
-    n
-  end
+    class << self
+      # main entry point
+      def load(robots_fn, dir = 'config/environments', host = nil)
+        # Validate parameters
+        robots_fn = File.join(dir, robots_fn) if dir
+        fail "FileNotFound: #{robots_fn}" unless File.file?(robots_fn)
 
-  # parse_lanes('') == ['default']
-  # parse_lanes(' ') == ['default']
-  # parse_lanes(' , ') == ['default']
-  # parse_lanes(' , ,') == ['default']
-  # parse_lanes('*') == ['*']
-  # parse_lanes('1') == ['1']
-  # parse_lanes('A') == ['A']
-  # parse_lanes('A , B') == ['A', 'B']
-  # parse_lanes('A,B,C') == ['A','B','C']
-  # parse_lanes('A-C,E') == ['A-C', 'E']
-  def parse_lanes(lanes_spec)
-    return ['default'] if lanes_spec.split(/,/).collect(&:strip).join('') == ''
-    lanes_spec.split(/,/).collect(&:strip).uniq
-  end
+        # read the YAML file
+        # puts "Loading #{robots_fn}"
+        robots =  YAML.load_file(robots_fn)
+        # puts robots
 
-  # build_queues('z','A') => ['z_A']
-  # build_queues('z','A,C') => ['z_A', 'z_C']
-  def build_queues(robot, lanes)
-    queues = []
-    parse_lanes(lanes).each do |i|
-      queues << [robot, i].join('_')
-    end
-    queues
-  end
+        # determine current host
+        host = `hostname -s`.strip unless host
+        # puts host
 
-  # main entry point
-  def load(robots_fn, dir = 'config/environments', host = nil)
-    # Validate parameters
-    robots_fn = File.join(dir, robots_fn) if dir
-    fail "FileNotFound: #{robots_fn}" unless File.file?(robots_fn)
+        # host = 'sul-robots1-dev' # XXX
+        fail "HostMismatch: #{host} not defined in #{robots_fn}" unless robots.include?(host) || robots.include?('*')
+        host = '*' unless robots.include?(host)
 
-    # read the YAML file
-    puts "Loading #{robots_fn}"
-    robots =  YAML.load_file(robots_fn)
-    # puts robots
+        parse_yaml(robots[host])
+      end
 
-    # determine current host
-    host = `hostname -s`.strip unless host
-    # puts host
+      # parse_instances(1) == 1
+      # parse_instances(16) == 16
+      # parse_instances(0) == 1
+      # parse_instances(99) => RuntimeError
+      def parse_instances(n)
+        fail "TooManyInstances: #{n} > #{ROBOT_INSTANCE_MAX}" if n > ROBOT_INSTANCE_MAX
+        n = 1 if n < 1
+        n
+      end
```

Comment:
The comments above this function demonstrate how it behaves, but they don't explain why it behaves this way.  So, what is the purpose of `parse_instances` because the purpose is not clear from the name of the function alone.